docs/03_goclient/15_sessions.md

```diff
@@ -0,0 +1,136 @@
+
+  // CreationTimeout: required, no default
+  //     Specifies how long session creation can take before returning an error
+  CreationTimeout  time.Duration
+}
+
+func CreateSession(ctx Context, sessionOptions *SessionOptions) (Context, error)
+```
+
+`CreateSession()` takes in workflow.Context, sessionOptions and returns a new context which contains metadata information of the created session (referred to as the **session context** below). When it's called, it will check the task list name specified in the `ActivityOptions` (or in the `StartWorkflowOptions` if the task list name is not specified in the `ActivityOptions`), and create the session on one of the workers which is polling that task list. 
+
+The returned session context should be used to execute all activities belonging to the session. The context will be cancelled if the worker executing this session dies or `CompleteSession()` is called. When using the returned session context to execute activities, a `workflow.ErrSessionFailed` error may be returned if the session framework detects that the worker executing this session has died. The failure of your activities won't affect the state of the session, so you still need to handle the errors returned from your activites and call `CompleteSession()` if necessary.
+
+`CreateSession()` will return an error if the context passed in already contains an open session. If all the workers are currently busy and unable to handle new sessions, the framework will keep retrying until the `CreationTimeout` you specified in the `SessionOptions` has passed before returning an error (check the **Concurrent Session Limitation** section for more details).
+
+```go
+func CompleteSession(ctx Context)
+```
+
+`CompleteSession()` releases the resources reserved on the worker, so it's important to call it as soon as you no longer need the session. It will cancel the session context and therefore all the activities using that session context. Note that it's safe to call `CompleteSession()` on a failed session, meaning that you can call it from a `defer` function after the session is successfully created.
+
+### Sample Code
+
+```go
+func FileProcessingWorkflow(ctx workflow.Context, fileID string) (err error) {
+  ao := workflow.ActivityOptions{
+    ScheduleToStartTimeout: time.Second * 5,
+    StartToCloseTimeout:    time.Minute,
+  }
+  ctx = workflow.WithActivityOptions(ctx, ao)
+
+  so := &workflow.SessionOptions{
+    CreationTimeout:  time.Minute,
+    ExecutionTimeout: time.Minute,
+  }
+  sessionCtx, err := workflow.CreateSession(ctx, so)
+  if err != nil {
+    return err
+  }
+  defer workflow.CompleteSession(sessionCtx)
+
+  var fInfo *fileInfo
+  err = workflow.ExecuteActivity(sessionCtx, downloadFileActivityName, fileID).Get(sessionCtx, &fInfo)
+  if err != nil {
+    return err
+  }
+
+  var fInfoProcessed *fileInfo
+  err = workflow.ExecuteActivity(sessionCtx, processFileActivityName, *fInfo).Get(ctx, &fInfoProcessed)
```

Comment:
For these three activities, you use `sessionCtx` for the Get method on the first one, but `ctx` for the other two. Is this correct?

Reply:
Thanks for the catch. It's my fault. I don't think it will cause any problem as I have tested the code locally. But it's always good to keep it consistent and avoid any confusion. I will update soon. Thanks.